contracts/Ethleverage.sol

```diff
@@ -106,55 +121,64 @@ contract Ethleverage {
 
   function leverage(uint256 _ethMarketPrice, uint256 _priceFloor) payable public returns (bool sufficient) {
     //TO-DO: w/ price floor or leverage ratio, determine the number of layers and LR
-    uint256  calcCR = (_ethMarketPrice.mul(makerLR)).div(_priceFloor);
-
+    uint256  calcCR = (_ethMarketPrice.mul(makerLR)).div(_priceFloor.mul(100));
 
+    //investor assignments
     Investor memory sender;
     sender = investors[msg.sender];
     investorAddresses.push(msg.sender);
-
     sender.layers = layers;
     sender.prinContr = msg.value;
     sender.CR = calcCR;
 
+    //external calls
+    ICDPContract tub = ICDPContract(CDPContract);
+    IwethContract weth = IwethContract(wethContract);
+
     uint recycledPeth;
     recycledPeth = sender.prinContr;
 
     // Step 3. Open CDPContract
-    bytes32 CDPInfo = ICDPContract(CDPContract).open();
+    bytes32 CDPInfo = tub.open();
     sender.cdps = CDPInfo;
+    CDPOpened(CDPInfo); //event
 
     // 1. convert eth into WETH
     //just in case the MakeGuy was wrong: wethContract.transfer(recycledPeth);
-    IwethContract(wethContract).transfer(recycledPeth);
-    uint DaiAmount = _priceFloor.div(makerLR);
+    weth.deposit.value(recycledPeth)();
+    uint DaiAmount = (_priceFloor.div(makerLR)).mul(100);
 
     for (uint i = 0; i < sender.layers; i++) {
         // 2a. Convert WETH into PETH
-        ICDPContract(CDPContract).join(recycledPeth);
-        //just in case the MakeGuy was wrong: pethContract.approve(address(this), recycledPeth);
+        tub.join(recycledPeth);
 
          // 4. deposit PETH into CDP
-        ICDPContract(CDPContract).lock(sender.cdps, recycledPeth);
+        tub.lock(sender.cdps, recycledPeth);
+        PethLocked(CDPInfo, recycledPeth); //event
 
          // 5. withdraw DAI
-        ICDPContract(CDPContract).draw(sender.cdps, DaiAmount); // may need to use liquidation ratio in this!
+        tub.draw(sender.cdps, DaiAmount); // may need to use liquidation ratio in this!
 
         //6. buy weth, sell dai
-        //OasisMarket.sellAllAmount(DaiContract, DaiAmount, WethContract, min_fill_amount)
-				recycledPeth = IMoneyMaker(moneyMakerKovan).buyAllEthWithDai().div(1e18);
-        //recycledPeth = wethReceived
+				//recycledPeth = IMoneyMaker(moneyMakerKovan).buyAllEthWithDai().div(1e18); //on mainNet
+        ILiquidator(liquidKovan).bust(DaiAmount); //for Kovan use only (NOTE: will not work if Total Liquidity Available is 0)
```

Comment:
If it won't won't in that scenario, we should have a check before we try and bust